### PR TITLE
ECOM-5031 create the serialization for programs api

### PR DESCRIPTION
**Pull request by schenedx**
@clintonb @rlucioni Please review
The PR creates the desired API for exposing programs data
Please note this PR is against @clintonb's program data model branch. Once the model branch is merged into master, I will update the PR base
The example program data this serialization provides is at https://gist.github.com/schenedx/9c8fa5b365e47c7795c6794b89b0afe1

### CODE REVIEW

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -182,7 +185,11 @@ def test_data(self):
             'marketing_slug': program.marketing_slug,
             'marketing_url': program.marketing_url,
             'card_image_url': program.card_image_url,
+            'banner_image_url': program.banner_image_url,
+            'authoring_organizations': OrganizationSerializer(program.authoring_organizations, many=True).data,
+            'courses': CourseSerializer(program.courses, many=True, context={'request': request}).data
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: Include a trailing comma.

**Comment by clintonb** [comment]
👍

**Comment by rlucioni** [comment]
You might want to rebase onto master since @clintonb's commits have merged.

File: course_discovery/apps/course_metadata/migrations/0011_auto_20160804_2331.py [hunk 2]
```diff
@@ -1,221 +0,0 @@
-# -*- coding: utf-8 -*-
-from __future__ import unicode_literals
-
-from django.db import migrations, models
-import django.db.models.deletion
-import django_extensions.db.fields
-import sortedm2m.fields
-
-
-class Migration(migrations.Migration):
```
Comment (rlucioni) [inline course_discovery/apps/course_metadata/migrations/0011_auto_20160804_2331.py]: Why is this migration being deleted?

File: course_discovery/apps/api/serializers.py [hunk 3]
```diff
@@ -50,8 +50,8 @@
 }
 
 PROGRAM_SEARCH_FIELDS = (
-    'text', 'uuid', 'title', 'subtitle', 'category', 'marketing_url', 'authoring_organizations', 'content_type',
-    'card_image_url', 'status',
+    'text', 'uuid', 'title', 'subtitle', 'category', 'marketing_url', 'organizations', 'content_type', 'status',
```
Comment (rlucioni) [inline course_discovery/apps/api/serializers.py:53]: Should this remain `authoring_organizations`? Why the change?

**Comment by rlucioni** [comment]
@schenedx I'm a little confused by the diff. You shouldn't have @clintonb's commits on this PR. Let me know if you need help resolving this. My suggestion is a cherry-pick of your changes onto a fresh branch.

**Comment by schenedx** [comment]
Closing this PR for https://github.com/edx/course-discovery/pull/225
